src/core/qgsshadingrenderer.h

```diff
@@ -0,0 +1,214 @@
+
+    /**
+     * Sets the \a strength of the eye dome lighting method.
+     *
+     * \see eyeDomeLightingStrength()
+     */
+    void setEyeDomeLightingStrength( double strength );
+
+    /**
+     * Returns the distance of the eye dome lighting method, that is the distance where the
+     * effect is apply from the source elevation.
+     *
+     * \see setEyeDomeLightingDistance()
+     */
+    double eyeDomeLightingDistance() const;
+
+    /**
+     * Sets the distance of the eye dome lighting method, that is the distance where the
+     * effect is apply from the source elevation.
+     *
+     * \see eyeDomeLightingDistance()
+     */
+    void setEyeDomeLightingDistance( double distance );
+
+    /**
+     * Returns the unit of the distance of the eye dome lighting method returned by eyeDomeLightingDistance().
+     *
+     * \see setEyeDomeLightingDistanceUnit()
+     */
+    const QgsUnitTypes::RenderUnit &eyeDomeLightingDistanceUnit() const;
+
+    /**
+     * Sets the unit of the distance of the eye dome lighting method set by setEyeDomeLightingDistance().
+     *
+     * \see eyeDomeLightingDistanceUnit()
+     */
+    void setEyeDomeLightingDistanceUnit( const QgsUnitTypes::RenderUnit &newEyeDomeLightingDistanceUnit );
+
+    //! Sets active the hillshading
+    void setActiveHillShading( bool active );
+
+    //! Returns whether the hillshading is active
+    bool isActiveHillShading() const;
+
+    /**
+     * Returns the z factor used by the hill shading method.
+     *
+     * \see setHillShadingZFactor(à)
+     */
+    double hillShadingZFactor() const;
+
+    /**
+     * Sets the z factor used by the hill shading method.
+     *
+     * \see hillShadingZFactor()
+     */
+    void setHillShadingZFactor( double zFactor );
+
+    /**
+     * Returns whether the hill shading method is multidirectional.
+     *
+     * \see setHillShadingMultidirectional()
+     */
+    bool isHillShadingMultidirectional() const;
+
+    /**
+     * Sets whether the hill shading method is multidirectional.
+     *
+     * \see isHillShadingMultidirectional()
+     */
+    void setHillShadingMultidirectional( bool multiDirectional );
+
+    /**
+     * Returns the altitude of the light (degree) that can be used by some methods (e.g. hill shading).
+     *
+     * \see setLightAltitude()
+     */
+    double lightAltitude() const;
+
+    /**
+     * Sets the altitude of the light (degree) that can be used by some methods (e.g. hill shading).
+     *
+     * \see setLightAltitude()
+     */
+    void setLightAltitude( double lightAltitude );
+
+    /**
+     * Returns the azimuth of the light (degree) that can be used by some methods (e.g. hill shading).
+     *
+     * \see setLightAltitude()
+     */
+    double lightAzimuth() const;
+
+    /**
+     * Sets the azimuth of the light (degree) that can be used by some methods (e.g. hill shading).
+     *
+     * \see lightAzimuth()
+     */
+    void setLightAzimuth( double lightAzimuth );
```

Comment:
I was wondering, and I preferred to have azimuth and altitude in the general context, for the future. Indeed, as it directional light parameters, it could be used by other methods that we will add.